chirho/robust/internals/linearize.py

```diff
@@ -90,7 +91,7 @@ def make_empirical_fisher_vp(
     batched_func_log_prob: Callable[[ParamDict, Point[T]], torch.Tensor] = torch.vmap(
         lambda p, data: func_log_prob(p, data, *args, **kwargs),
         in_dims=(None, 0),
-        randomness="different",
+        randomness="same",
```

Comment:
Could you explain this change? IIUC this will reuse randomness within `NMCLogPredictiveLikelihood` across test data samples in a batch, which will affect the statistical properties of our Fisher matrix estimator.

Reply:
I made this change because we no longer need randomness to change with the Hessian formulation of Fisher in #428. In #428, we can approximate Fisher with

$$
-\sum_{n=1}^N \nabla^2_{\phi} \log \sum_{t=1}^T p(x_n \mid \theta_t), \ \theta_t \sim q(\theta \mid \phi).
$$ 

Since there's only one call to $p(x_n \mid \theta_t)$, we can share randomness across datapoints unlike in the score formulation of Fisher. However, I guess it doesn't really hurt to have randomness even in the Hessian formulation case. So I'll move it back to "different".